### ISSUE TITLE
Cannot `allow(clippy::disallowed_macros)` inside of a macro expansion?

### ISSUE THREAD

**Issue by BatmanAoD** [rust-lang/rust-clippy#15312]
### Description

I wanted to write a "wrapper" macro around `tracing::error`, then disallow `tracing::error` itself, but it seems I can't just put an `allow` in the "wrapper" macro itself. Here's a sort of "dummy" version of what I'm doing:

```rust
macro_rules! my_error {
    ($($args:tt)+) => {
        #[allow(clippy::disallowed_macros)]
        if <some logic> {
            tracing::error!(<some extra fields>, $($args)+);
        } else {
            tracing::error!($($args)+)
        }
    }
}
```

The `allow` seems to have absolutely no effect. Is there a way to do this correctly?

### Version

```text
rustc 1.88.0 (6b00bc388 2025-06-23)
binary: rustc
commit-hash: 6b00bc3880198600130e1cf62b8f8a93494488cc
commit-date: 2025-06-23
host: aarch64-apple-darwin
release: 1.88.0
LLVM version: 20.1.5
```

### Additional Labels

@rustbot label +C-question

**Comment by profetia** [comment]
I cannot reproduce this issue. Here is what I did:
main.rs:
```rust
  macro_rules! my_error {
      ($($args:tt)+) => {
          #[allow(clippy::disallowed_macros)]
          if todo!() {
              std::eprintln!($($args)+);
          } else {
              std::eprintln!($($args)+);
          }
      }
  }

  pub fn test() {
      my_error!("This is an error message with some extra fields");
  }

```

clippy.toml
```toml
disallowed-macros = ["std::eprintln"]
```

**Comment by BatmanAoD** [comment]
I don't know why, but it seems to be specific to `tracing::error!`. Please try with that.

**Comment by Alexendoo** [comment]
It will occur when the macro emits an attribute

```rust
macro_rules! m {
    () => {
        #[repr(C)]
        struct S {}
    };
}

#[allow(clippy::disallowed_macros)]
fn main() {
    m!();
}
```

**Comment by BatmanAoD** [comment]
Interesting; any idea if there's a possible workaround?

**Comment by Alexendoo** [comment]
The lint emits at the crate root, so you could do

```rust
#![allow(clippy::disallowed_macros)]

#[warn(clippy::disallowed_macros)]
mod foo;
```

which of course is annoying the more items you have in there, but if you only need the lint in one/two specific modules it could work

**Comment by zihan0822** [comment]
same issue as [#14017](https://github.com/rust-lang/rust-clippy/issues/14017)
